data/api.ts

```diff
@@ -12,6 +12,6 @@ export async function postEndpoint(url: string, data: any) {
   return fetchEndpoint(url, 'POST', data)
 }
 
-export async function putEndpoint(url: string, data: any) {
+export async function putEndpoint(url: string, data?: any) {
```

Comment:
As a PUT request is usually expected to have a body, we should probably include a `Content-Length: 0` header to clarify that the body is expected to be empty.